Keas.Mvc/ClientApp/components/Equipment/EquipmentBigFixInfo.tsx

```diff
@@ -0,0 +1,121 @@
+import * as PropTypes from "prop-types";
+import * as React from "react";
+import { Button, Modal, ModalBody, ModalFooter, ModalHeader, Table } from "reactstrap";
+
+interface IProps {
+    bigfixId: string;
+}
+
+interface IState {
+    bigfixModal: boolean;
+    computerInfo: object;
+    isFetched: boolean;
+}
+
+export default class EquipmentDetails extends React.Component<IProps, IState> {
+    public static contextTypes = {
+        fetch: PropTypes.func,
+        permissions: PropTypes.array,
+        router: PropTypes.object,
+        team: PropTypes.object
+    };
+
+    public state = { bigfixModal: false, computerInfo: {}, isFetched: false };
+
+    public modalToggle = () => {
+        this.setState({
+            bigfixModal: !this.state.bigfixModal
+        });
```

Comment:
If your new state depends on the old state, you should use the functional form of set state otherwise you can get weird bugs.  ex:
 
```
this.setState(prevState => ({
  showModal: !prevState.showModal
}));
```

Reply:
fixed